src/asn1_types/oid.rs

```diff
@@ -13,12 +13,27 @@ use num_traits::Num;
 
 /// An error for OID parsing functions.
 #[derive(Debug)]
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
 pub enum OidParseError {
+    #[cfg_attr(
+        feature = "std",
+        error("Non-relative OIDs must be at least 2 components long")
+    )]
     TooShort,
+
+    #[cfg_attr(feature = "std", error("Relative OIDs must not be empty"))]
+    RelativeTooShort,
```

Comment:
Could this be implemented without adding the new variant? It's a semver breaking change with this approach and it feels like the value of the separate `RelativeTooShort` variant might not justify it. WDYT?

Reply:
Right I forgot to keep semver in mind. Of course we don't need to introduce a breaking change for this.

Fixed it.